external/llvm-project/mlir/lib/Target/LLVM/ROCDL/Target.cpp

```diff
@@ -271,10 +273,9 @@ void SerializeGPUModuleBase::addControlVariables(
                        isaVersion.Minor + 100 * isaVersion.Stepping +
                            1000 * isaVersion.Major,
                        32);
-    int abi = 500;
-    abiVer.getAsInteger(0, abi);
     addControlVariable("__oclc_ABI_version", abi, 32);
   }
+  addControlVariable(".amdhsa_code_object_version", abi, 32);
```

Comment:
Nope!

We want a module flag

And this is independent of device library loading

Reply:
And this won't run if no device libs are being loaded per the if statement at the top of this function